docs/openfaas-pro/sns-events.md

```diff
@@ -0,0 +1,89 @@
+
+
+### Install the connector with Helm
+
+* Set up AWS
+    You can configure permissions using a dedicated IAM user, or if your cluster is configured for AWS IAM, ambient credentials mapped into the Pod at runtime.
+
+* Set up the connector
+    The SNS connector can be installed using its [helm chart](https://github.com/openfaas/faas-netes/tree/master/chart/sns-connector)
+
+
+* Configure the connector for you needs
+        
+    `callbackURL` - Public callback URL for subscriptions
+
+    `topicARN` - SNS topic ARN
+
+    `awsRegion` - The AWS region for your SNS topics
+
+> Note that the callback URL has to be publicly accessible to receive SNS messages. An ingress configuration example is available in the [chart README](https://github.com/openfaas/faas-netes/tree/master/chart/sns-connector#configure-ingress)
+
+## Usage
+
+Once you configured the connector to receive callbacks for an SNS topic you can annotate you functions so they get triggered by any incoming message on that topic.
+
+```bash
+faas-cli deploy \
+    --name printer \
+    --annotation topic=arn:aws:sns:us-east-1:222943518635:of-event
+```
+
+The printer function will print out the body and headers of each request which makes it a great function for debugging events.
+
+You can publish a message to your topic from the AWS console. 
+
+Example output of the printer function:
+
+```bash
+faas-cli logs printer
+
+2023-01-13T15:05:46Z User-Agent=[Go-http-client/1.1]
+2023-01-13T15:05:46Z Accept-Encoding=[gzip]
+2023-01-13T15:05:46Z 2023/01/13 15:05:46 POST / - 202 Accepted - ContentLength: 0B (0.0004s)
+2023-01-13T15:05:46Z X-Forwarded-Uri=[/function/printer.openfaas-fn]
+2023-01-13T15:05:46Z X-Topic=[arn:aws:sns:us-east-1:222943518635:of-event]
+2023-01-13T15:05:46Z X-Call-Id=[c003629d-81dc-4d7a-bacd-96bb2a29782c]
+2023-01-13T15:05:46Z X-Sns-Attr-Count=[20]
+2023-01-13T15:05:46Z X-Sns-Attr-Foo-Type=[String]
+2023-01-13T15:05:46Z X-Connector=[connector-sdk openfaasltd/sns-connector]
+2023-01-13T15:05:46Z X-Forwarded-Host=[gateway.cromulon.local:8000]
+2023-01-13T15:05:46Z X-Inlets-Id=[2aaea42757464a93bdaebc76d832cfea]
+2023-01-13T15:05:46Z X-Sns-Arn=[arn:aws:sns:us-east-1:222943518635:of-event]
+2023-01-13T15:05:46Z X-Sns-Attr-Foo=[bar]
+2023-01-13T15:05:46Z X-Start-Time=[1673622346748456857]
+2023-01-13T15:05:46Z X-Forwarded-For=[192.168.195.37]
+2023-01-13T15:05:46Z X-Sns-Attr-Count-Type=[Number]
+2023-01-13T15:05:46Z X-Sns-Message-Id=[6f0ee500-756a-5391-9742-b93b00e0afb0]
+2023-01-13T15:05:46Z X-Sns-Subject=[openfaas]
+2023-01-13T15:05:46Z 
+2023-01-13T15:05:46Z Hello OpenFaaS
+```
+
+Additional headers are made available on the request. These headers contain some SNS message metadata and the message attributes.
+
+* `X-Sns-Message-Id` - the message id.
+* `X-Sns-Subject` - the message subject.
+* `X-Sns-Arn` - the arn of the SNS topic.
+* `X-Sns-Attr-<key>` - value of an SNS message attribute.
+* `X-Sns-Attr-<Key>-Type` - type of the SNS message attribute.
+
```

Comment:
So for example, if you added an attributed called X with content of Y, you'd get the following two extra headers:

...